src/org-creator/orgCreator.controller.ts

```diff
@@ -0,0 +1,38 @@
+import { Controller, Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { EventPattern, Payload } from '@nestjs/microservices';
+import * as jwt from 'jsonwebtoken';
+import { Logger } from '../logger/logger.service';
+import { ClaimRequestEventDto } from './orgCreator.dto';
+import { IClaimToken } from './orgCreator.type';
+
+@Injectable()
+@Controller()
+export class OrgCreatorController {
+  constructor(
+    private configService: ConfigService,
+    private readonly logger: Logger,
+  ) {
+    this.logger.setContext(OrgCreatorController.name);
+  }
+
+  @EventPattern('*.claim.exchange')
+  async createOrg(@Payload() message: ClaimRequestEventDto) {
+    const { token } = message;
+    const {
+      payload: { claimData },
+    } = jwt.decode(token, { complete: true });
+
+    const permittedEventRole = this.configService.get<string>(
+      'PERMITTED_ORG_CREATOR_ROLE',
+    );
+    if (claimData?.claimType !== permittedEventRole) {
```

Comment:
hmmm, for the claim request creation event we'll probably always have `permittedEventRole` because the DTO validation would ensure that even if it's an empty string `' '`.  so even if  `PERMITTED_ORG_CREATOR_ROLE ` is null/undefined, `undefined !== ' '` would always be true.  It's a valid edge case to have though. I'll add it in the test.

Reply:
Shouldn't we rely on cache-server for org creator role verification?

Reply:
@JGiter I don't see how the cache-server helps here, to be honest. Could you elaborate on how you see the cache-server playing a role please?

Reply:
Why not ask claims of requester and check if there is approved claim with PERMITTED_ORG_CREATOR?

Reply:
@JGiter so at that point, the claims request would not have been approved so there probably won't be an approved claim to check for.